### PR TITLE
Add BisectBatchOnFunctionError option for Dynamo and Kinesis event sourcing

### LINKED ISSUE #7041: expose Bisect on Function Error prop for Kinesis Streams
# Feature Proposal

https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-supports-failure-handling-features-for-kinesis-and-dynamodb-event-sources/

## Description

1. What is the use case that should be solved. The more detail you describe this in the easier it is to understand for us.
1. **Optional:** If there is additional config how would it look

Similar or dependent issues:

- https://github.com/serverless/serverless/pull/7024

### PR DESCRIPTION
**Pull request by jchaasis**
What did you implement

Add the ability to configure the BisectBatchOnFunctionError feature for failure handling on Kinesis and DynamoDB streams.

Closes #7041 

## How can we verify it

Sample serverless.yml:

```yml
functions:
  preprocess:
    handler: handler.preprocess
    events:
      - stream:
          arn: arn:aws:kinesis:region:XXXXXX:stream/foo
          bisectBatchOnFunctionError: true
```

## Todos

<details>
<summary>Useful Scripts</summary>

- `npm run test:ci` --> Run all validation checks on proposed changes
- `npm run lint:updated` --> Lint all the updated files
- `npm run lint:fix` --> Automatically fix lint problems (if possible)
- `npm run prettier-check:updated` --> Check if updated files adhere to Prettier config
- `npm run prettify:updated` --> Prettify all the updated files

</details>

- [x] Write and run all tests
- [x] Write documentation
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: lib/plugins/aws/package/compile/events/stream/index.js [hunk 1]
```diff
@@ -203,6 +203,11 @@ class AwsCompileStreamEvents {
               streamResource.Properties.MaximumRetryAttempts = event.stream.maximumRetryAttempts;
             }
 
+            if (event.stream.bisectBatchOnFunctionError) {
+              streamResource.Properties.BisectBatchOnFunctionError =
+                event.stream.bisectBatchOnFunctionError;
```
Comment (medikoo) [inline lib/plugins/aws/package/compile/events/stream/index.js]: It'll be good to ensure in all cases it's a _boolean_ that ends in CF template, so I'd just assign `true`

Comment (jchaasis) [reply lib/plugins/aws/package/compile/events/stream/index.js]: Thanks for the review! That sounds good, I'll make the adjustment 👍

**Comment by neverendingqs** [comment]
@medikoo - bump :). Hoping it can make it into `16.1.0` if possible.

**Comment by walenzack** [comment]
I see this has been dropped from `1.61.0`. Any tips on when will it be available?

**Comment by medikoo** [comment]
@walenzack it'll be published with v1.61.0 most likely on Monday

**Pull request merged**